### DISCUSSION TITLE
Thoughts about Launcher application ??

### DISCUSSION THREAD

**Discussion by U-C-S** [U-C-S/Hurl#132] [General]
Maybe a weird name for an application that acts as a proxy between the hurl's browser selector and other applications that trigger it.

This proxy application initially started as a personal weekend experiment to try out IPC (#123), ended up with full working implementation of something useful. It does
- Launch Hurl's Browser Selector window
- Launches it really fast... 
- Provides better communication with the Browser extension.

In future, Can be extended to handling 
- Rules checks (or should It ?)
- Timed Default
- even faster than ever before

### Why ?
- Each time, an application triggers hurl, a fresh instance of hurl is started which immediately checks if another instance of hurl is running and forwards current context to an already existing instance if it exists 
- which is quite slow with how many deps and Hurl written in C#/.NET, which needs a extra JIT compilation step before running it.
- This brought the need for some language that compiles ahead of time and runs immediately.
- Rust, C++ are my choices and Rust wins for my familiarity with its ecosystem.

Using Named Pipes for IPC, Hurl now responds super instantaneously.

### Cons
- Complicates the build system (looking for a way to simplify this, maybe Bazel?)
- Requires a introduction of new language, this making harder to setup for new contributors
- Portability
- The inter-dependency btw launcher and hurl application, makes it prone to new kinds of errors

That's all about it I believe. Use this discussion to post your thoughts about this recent introduction.

**Comment by U-C-S** [comment]
@Jay-o-Way ?? Any thoughts on this since you have your PRs 

I'll update about development docs soon.

**Reply by Jay-o-Way** [reply]
Oh, this area is not for me. Sorry 😇
